openslides_backend/action/actions/user/password_mixins.py

```diff
@@ -0,0 +1,47 @@
+from typing import Any, Callable, Dict
+
+from openslides_backend.shared.patterns import fqid_from_collection_and_id
+
+from ....shared.exceptions import ActionException
+from ...action import Action
+from ...util.typing import ActionData
+
+
+class SetPasswordMixin(Action):
+    def reset_password(self, instance: Dict[str, Any]) -> None:
+        instance["password"] = instance["default_password"]
+        self.set_password(instance)
+
+    def set_password(self, instance: Dict[str, Any]) -> None:
+        """
+        Hashes the password given in the instance (which is assumed to be plain text) and sets it as
+        the default password if `set_as_default` is True in the instance.
+        """
+        if "meta_new" not in instance:
+            user = self.datastore.get(
+                fqid_from_collection_and_id("user", instance["id"]),
+                ["saml_id"],
+                lock_result=False,
+            )
+            if user.get("saml_id"):
+                raise ActionException(
+                    f"user {user['saml_id']} is a Single Sign On user and has no local Openslides passwort."
+                )
+
+        password = instance.pop("password")
```

Comment:
IMO you should check, whether the saml_id is in the instance. Callers could never be sure, who is responsible for this, especially because you check this on database.

Reply:
done